lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -814,4 +970,148 @@ class ProviderTest {
         assertNull(receipt.actualFee)
         assertNull(receipt.failureReason)
     }
+
+    @Test
+    fun `get nonce with block number`() {
+        val provider = rpcProvider()
+        val request = provider.getNonce(contractAddress, latestBlock.number)
+        val response = request.send()
+
+        assertNotNull(response)
+    }
+
+    @Test
+    fun `get nonce with block hash`() {
+        val provider = rpcProvider()
+        val request = provider.getNonce(contractAddress, latestBlock.hash)
+        val response = request.send()
+
+        assertNotNull(response)
+    }
+
+    @Test
+    fun `get block with transactions with block tag`() {
+        // FIXME: We should also test for 'pending' tag, but atm they are not supported in devnet
+        val provider = rpcProvider()
+        val request = provider.getBlockWithTxs(BlockTag.LATEST)
+        val response = request.send()
+
+        assertNotNull(response)
+        assertTrue(response is BlockWithTransactionsResponse)
```

Comment:
Why is getNonce only tested in rpc? I believe gateway also has an endpoint for getting a nonce.